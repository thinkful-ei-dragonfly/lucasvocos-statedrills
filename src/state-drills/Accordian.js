import React from 'react';

class Accordian extends React.Component{
  static defaultProps = {
    sections: []
  }
  state = {
    currentTabIndex: null
  }
  handleClick(index){
    this.setState({ currentTabIndex: index})
  }
  renderButtons(){
    return this.props.sections.map((tab, index) => (
      <li key={index} onClick={()=> this.handleClick(index)}>
        {tab.title}
        {(this.state.currentTabIndex === index) && this.renderContent()}
      </li>
    ))
  }
  renderContent(){
    const currentTab = this.props.sections[this.state.currentTabIndex];

    return (
      <p className="content">
        {currentTab.content}
      </p>
    )
  }
  render(){
    return (
      <ul>
        {this.renderButtons()}
      </ul>
    )
  }
}
export default Accordian;
