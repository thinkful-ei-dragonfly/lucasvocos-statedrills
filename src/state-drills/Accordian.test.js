import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordian from './Accordian'

describe('Testing Accordian', () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]
  it('renders without errors [Smoke Test]', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders tabs but none are active', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('opens any clicked li', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('li').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('sets last clicked element as active', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('li').at(1).simulate('click')
    wrapper.find('li').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
