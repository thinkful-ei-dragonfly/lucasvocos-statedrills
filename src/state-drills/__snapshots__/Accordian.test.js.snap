// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Accordian opens any clicked li 1`] = `
<ul>
  <li
    key="0"
    onClick={[Function]}
  >
    Section 1
  </li>
  <li
    key="1"
    onClick={[Function]}
  >
    Section 2
    <p
      className="content"
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
    onClick={[Function]}
  >
    Section 3
  </li>
</ul>
`;

exports[`Testing Accordian renders empty given no tabs 1`] = `<ul />`;

exports[`Testing Accordian renders tabs but none are active 1`] = `
<ul>
  <li
    key="0"
    onClick={[Function]}
  >
    Section 1
  </li>
  <li
    key="1"
    onClick={[Function]}
  >
    Section 2
  </li>
  <li
    key="2"
    onClick={[Function]}
  >
    Section 3
  </li>
</ul>
`;

exports[`Testing Accordian sets last clicked element as active 1`] = `
<ul>
  <li
    key="0"
    onClick={[Function]}
  >
    Section 1
  </li>
  <li
    key="1"
    onClick={[Function]}
  >
    Section 2
  </li>
  <li
    key="2"
    onClick={[Function]}
  >
    Section 3
    <p
      className="content"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;
